src/hotspot/share/compiler/compilerOracle.cpp

```diff
@@ -341,7 +341,48 @@ bool CompilerOracle::has_option_value(const methodHandle& method, enum CompileCo
   return false;
 }
 
+static bool resolve_inlining_predicate(enum CompileCommand option, const methodHandle& method) {
+  assert(option == CompileCommand::Inline || option == CompileCommand::DontInline, "Sanity");
+  bool v1 = false;
+  bool v2 = false;
+  bool has_inline = CompilerOracle::has_option_value(method, CompileCommand::Inline, v1);
+  bool has_dnotinline = CompilerOracle::has_option_value(method, CompileCommand::DontInline, v2);
+  if (has_inline && has_dnotinline) {
+    if (v1 && v2) {
+      // Conflict options detected
+      // Find the last one for that method and return the predicate accordingly
+      enum CompileCommand last_one = CompileCommand::Unknown;
+      TypedMethodOptionMatcher* current = option_list;
```

Comment:
Add comment that `option_list` lists options in reverse order. So the first option we find is the last which was specified.

Reply:
> Add comment that `option_list` lists options in reverse order. So the first option we find is the last which was specified.

Updated.
Thanks.

Reply:
if has_inline == true && has_dnotinline == true, but v1 == false && v2 == false,  two options conflict, don't  they? 
in this case, you go to else, which actually bias to Inline instead of the last seen option.

Reply:
maybe like this:

Reply:
> if has_inline == true && has_dnotinline == true, but v1 == false && v2 == false, two options conflict, don't they?

I don't think they are conflict.

This is because
```
-XX:CompilerCommand=inline,foo,false     ==> doesn't mean to disallow the inline of foo
-XX:CompilerCommand=dontinline,foo,false ==>  doesn't mean to force inline of foo
```

So if `inline` and `dontinline` options are set to false, they actually have no effect to the compiler.
Thanks.

Reply:
make sense. I take back my comment.  those 2 options don't have any effect when they are false.